LabAssignments/LabAssignment3/LabAssignment3.ipynb

```diff
@@ -331,13 +356,1801 @@
     "split_indices = cross_validator.split(pca_feature, labels)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train and test the model using different feature engineering methods to verify performance improvement\n",
+    "\n",
+    "# Use original data\n",
+    "X = features\n",
+    "y = labels\n",
+    "\n",
+    "# Use PCA data without cross validation\n",
+    "X = pca_feature\n",
+    "y = labels\n",
+    "\n",
+    "# Use PCA data with cross validation\n",
+    "# binary classification data\n",
+    "for train_indices, test_indices in split_indices:\n",
+    "    X_train, y_train = pca_feature[train_indices], (labels[train_indices] >1)\n",
```

Comment:
Similarly, each iteration, the X_train and y_train will be overwritten by next iteration.  
if you are not going to use cross-validation, you can use something like below and remove the loop.
X_train, y_train = pca_feature[split_indices[0][0]], pca_feature[split_indices[0][1]]